helix-term/tests/test/movement.rs

```diff
@@ -86,6 +86,25 @@ async fn cursor_position_newly_opened_file() -> anyhow::Result<()> {
     Ok(())
 }
 
+#[tokio::test]
+async fn cursor_position_append_eof() -> anyhow::Result<()> {
+    test((
+        "#[f|]#oo",
+        "eabar<esc>",
+        helpers::platform_line("#[foobar|]#\n").as_ref()
+    ))
+    .await?;
+    
+    test((
+        "foo#[|]#",
+        "babar<esc>",
+        helpers::platform_line("#[foobar|]#\n").as_ref()
+    ))
+    .await?;
```

Comment:
I think the `e` and `b` obscure the test cases here a little bit. Instead could we do those in the setup?

```rs
// Selection is forwards
test((
    "#[foo|]#",
    "abar<esc>",
    helpers::platform_line("#[foobar|]#\n").as_ref(),
))
.await?;

// Selection is backwards
test((
    "#[|foo]#",
    "abar<esc>",
    helpers::platform_line("#[foobar|]#\n").as_ref(),
))
.await?;
```